core/test/unit/models/base/index_spec.js

```diff
@@ -153,4 +153,174 @@ describe('Models: base', function () {
+            const forgeStub = sandbox.stub(models.Base.Model, 'forge')
+                .returns(model);
+            const fetchStub = sandbox.stub(model, 'fetch')
+                .resolves(fetchedModel);
+
+            const findOneReturnValue = models.Base.Model.findOne(data, unfilteredOptions);
+
+            should.equal(findOneReturnValue, fetchStub.returnValues[0]);
+
+            return findOneReturnValue.then((result) => {
+                should.equal(result, fetchedModel);
+
+                should.equal(filterOptionsSpy.args[0][0], unfilteredOptions);
+                should.equal(filterOptionsSpy.args[0][1], 'findOne');
+
+                should.equal(filterDataSpy.args[0][0], data);
+
+                const filteredData = filterDataSpy.returnValues[0];
+                should.deepEqual(forgeStub.args[0][0], filteredData);
+
+                const filteredOptions = filterOptionsSpy.returnValues[0];
+                should.equal(fetchStub.args[0][0], filteredOptions);
+            });
+        });
+    });
+
+    describe('static edit(data, unfilteredOptions)', function () {
+        it('resolves with the savedModel after forges model w/ id, fetches w/ filtered options, saves w/ filtered data and options and method=update', function () {
+            const data = {
+                life: 'suffering'
+            };
+            const unfilteredOptions = {
+                id: 'something real special',
+            };
+            const model = models.Base.Model.forge({});
+            const savedModel = models.Base.Model.forge({});
+            const filterOptionsSpy = sandbox.spy(models.Base.Model, 'filterOptions');
+            const filterDataSpy = sandbox.spy(models.Base.Model, 'filterData');
+            const forgeStub = sandbox.stub(models.Base.Model, 'forge')
+                .returns(model);
+            const fetchStub = sandbox.stub(model, 'fetch')
+                .resolves(model);
+            const saveStub = sandbox.stub(model, 'save')
+                .resolves(savedModel);
+
+            return models.Base.Model.edit(data, unfilteredOptions).then((result) => {
+                should.equal(result, savedModel);
+
+                should.equal(filterOptionsSpy.args[0][0], unfilteredOptions);
+                should.equal(filterOptionsSpy.args[0][1], 'edit');
+
+                should.equal(filterDataSpy.args[0][0], data);
+
+                const filteredOptions = filterOptionsSpy.returnValues[0];
+                should.deepEqual(forgeStub.args[0][0], {id: filteredOptions.id});
+
+                should.equal(fetchStub.args[0][0], filteredOptions);
+
+                const filteredData = filterDataSpy.returnValues[0];
+                should.equal(saveStub.args[0][0], filteredData);
+                should.equal(saveStub.args[0][1].method, 'update');
+                should.deepEqual(_.omit(saveStub.args[0][1], 'method'), filteredOptions);
```

Comment:
This is annoying and it's because we do `_.merge({method: 'update', options})` in the edit method however in the add method we simply do `options.method = 'insert'`. From what I can tell, method isn't allowed to be passed in anyway (might be wrong), which means when reading the edit method although it might look like you can override the method, you can't - so should I update the edit method to just assign it?

Reply:
Do you mean?

```
if (object) {
  options.method = 'update';
  return object.save(data, options);
}
```

Reply:
Yeah :+1: